shared/src/uniswap_pool.rs

```diff
@@ -19,10 +20,72 @@ pub trait PoolFetching: Send + Sync {
     async fn fetch(&self, token_pairs: HashSet<TokenPair>) -> Vec<Pool>;
 }
 
-#[derive(Clone)]
+#[derive(Clone, Hash)]
 pub struct Pool {
     pub tokens: TokenPair,
     pub reserves: (u128, u128),
+    pub fee: Rational,
```

Comment:
Rational is Ratio<isize>. The fee has to be positive so we could use Ratio<usize> or Ratio<u32>. Just pointing out, not super important.